bpf/bpf_lxc.c

```diff
@@ -1119,19 +1119,22 @@ ipv6_policy(struct __ctx_buff *ctx, int ifindex, __u32 src_label, __u8 *reason,
 					   verdict, policy_match_type, audited);
 	}
 
-	if (ret == CT_NEW) {
 #ifdef ENABLE_DSR
-	{
+	if (ret == CT_NEW || ret == CT_REOPENED) {
 		bool dsr = false;
 
 		ret = handle_dsr_v6(ctx, &dsr);
 		if (ret != 0)
 			return ret;
 
-		ct_state_new.dsr = dsr;
+		if (ret == CT_NEW)
```

Comment:
I would set the dsr flag regardless whether it was `CT_NEW` or `CT_REOPENED`. Otherwise, `ct_state` is not correct wrt to DSR if consumed later in the code.

Reply:
We use ```ct_state_new``` late in the code only if it`s ```CT_NEW``` state, but if it's  ```CT_REOPENED``` state, ```ct_state_new``` is not consumed later in the code. I understand your concern, if you insist I can change it.

Reply:
Let's change it.

Reply:
I rechecked the code and realized that I had overwritten the state:
`		ret = handle_dsr_v6(ctx, &dsr);
		if (ret != 0)
			return ret;
`
I fix it up also.